src/main/java/net/snowflake/client/core/SFBaseSession.java

```diff
@@ -202,93 +219,201 @@ public void setSessionId(String sessionId) {
     return databaseMajorVersion;
   }
 
+  /**
+   * Set database major version
+   *
+   * @param databaseMajorVersion the database major version
+   */
   public void setDatabaseMajorVersion(int databaseMajorVersion) {
     this.databaseMajorVersion = databaseMajorVersion;
   }
 
+  /**
+   * Get the database minor version
+   *
+   * @return database minor version
+   */
   public int getDatabaseMinorVersion() {
     return databaseMinorVersion;
   }
 
+  /**
+   * Set the database minor version
+   *
+   * @param databaseMinorVersion the minor version
+   */
   public void setDatabaseMinorVersion(int databaseMinorVersion) {
     this.databaseMinorVersion = databaseMinorVersion;
   }
 
+  /**
+   * Get if prepared statement logging is enabled
+   *
+   * @return true if enabled
+   */
   public boolean getPreparedStatementLogging() {
     return this.preparedStatementLogging;
   }
 
+  /**
+   * Set prepared statement logging
+   *
+   * @param value boolean
+   */
   public void setPreparedStatementLogging(boolean value) {
     this.preparedStatementLogging = value;
   }
 
+  /**
+   * Get inject file upload failure
+   *
+   * @return file to fail
+   */
   public String getInjectFileUploadFailure() {
     return this.injectFileUploadFailure;
   }
 
+  /**
+   * Set inject file upload failure
+   *
+   * @param fileToFail the file to fail
+   */
   public void setInjectFileUploadFailure(String fileToFail) {
     this.injectFileUploadFailure = fileToFail;
   }
 
+  /**
+   * Get timestamp mapped type
+   *
+   * @return SnowflakeType
+   */
   public SnowflakeType getTimestampMappedType() {
     return timestampMappedType;
   }
 
+  /**
+   * Set the timestamp mapped type
+   *
+   * @param timestampMappedType SnowflakeType
+   */
   public void setTimestampMappedType(SnowflakeType timestampMappedType) {
     this.timestampMappedType = timestampMappedType;
   }
 
+  /**
+   * Get if result column is case-insensitive
+   *
+   * @return true if result column is case-insensitive
+   */
   public boolean isResultColumnCaseInsensitive() {
     return isResultColumnCaseInsensitive;
   }
 
+  /**
+   * Set if result column is case-insensitive
+   *
+   * @param resultColumnCaseInsensitive boolean
+   */
   public void setResultColumnCaseInsensitive(boolean resultColumnCaseInsensitive) {
     isResultColumnCaseInsensitive = resultColumnCaseInsensitive;
   }
 
+  /**
+   * Check if we want to treat decimal as int JDBC types
+   *
+   * @return true if decimal is treated as int
+   */
   public boolean isJdbcTreatDecimalAsInt() {
     return isJdbcTreatDecimalAsInt;
   }
 
+  /**
+   * Set if decimal should be treated as int type
+   *
+   * @param jdbcTreatDecimalAsInt boolean
+   */
   public void setJdbcTreatDecimalAsInt(boolean jdbcTreatDecimalAsInt) {
     isJdbcTreatDecimalAsInt = jdbcTreatDecimalAsInt;
   }
 
+  /**
+   * Check if decimal should be treated as int for arrow types
+   *
+   * @return true if decimal should be treated as int
```

Comment:
for arrow types - we can remove comment describing what methods do and put it in `@return`